packages/article-comments/src/comments.js

```diff
@@ -147,7 +150,13 @@ class Comments extends Component {
         executeSSOtransaction(() => {});
       } else {
         document.addEventListener("spot-im-api-ready", () => {
-          executeSSOtransaction(() => {});
+          if (isFeatureFlagEnabled) {
+            if (shouldReauthenticateUser()) {
+              executeSSOtransaction();
+            }
+          } else {
+            executeSSOtransaction(() => {});
+          }
```

Comment:
It would be a bit cleaner to do something like this since the executeSSoTransaction will fire regardless ```suggestion
          if (isFeatureFlagEnabled) {
            shouldReauthenticateUser()
            }

            executeSSOtransaction(() => {});
          
```